tsl/src/nodes/columnar_scan/columnar_scan.c

```diff
@@ -400,6 +461,42 @@ should_project(const CustomScanState *state)
 #endif
 }
 
+static inline bool
+should_check_segmentby_qual(ScanDirection direction, const TupleTableSlot *slot)
+{
+	if (likely(direction == ForwardScanDirection))
+		return arrow_slot_is_first(slot);
+
+	Assert(direction == BackwardScanDirection);
+	return arrow_slot_is_last(slot);
```

Comment:
Is it possible to cover this line in a test as well?

Reply:
Should be covered now.